lib/src/model_utils.dart

```diff
@@ -108,49 +109,67 @@ bool hasPrivateName(Element e) {
       return true;
     }
   }
-  if (e is LibraryElement &&
-      (e.identifier.startsWith('dart:_') ||
-          e.identifier.startsWith('dart:nativewrappers/') ||
-          ['dart:nativewrappers'].contains(e.identifier))) {
-    return true;
-  }
   if (e is LibraryElement) {
+    if (e.identifier.startsWith('dart:_') ||
+        e.identifier.startsWith('dart:nativewrappers/') ||
+        'dart:nativewrappers' == e.identifier) {
+      return true;
+    }
     var elementLocation = e.location;
     if (elementLocation != null) {
-      var locationParts = elementLocation.components[0].split(slashes);
+      var elementUri = elementLocation.components[0];
       // TODO(jcollins-g): Implement real cross package detection
-      if (locationParts.length >= 2 &&
-          locationParts[0].startsWith('package:') &&
-          locationParts[1] == 'src') return true;
+      if (_uriIsPackageImplementation(elementUri)) {
+        return true;
+      }
     }
   }
   return false;
 }
 
+/// Returns whether [uri] is a package's implementation, a URI starting with
+/// "package:", and with a first path segment of "src".
+bool _uriIsPackageImplementation(String uri) {
```

Comment:
And use `Uri` here, and `Uri.pathSegments`. If we can use standard URI implementation, we should, unless there are good reasons.

Reply:
Done